bugsnag-android-performance/src/main/kotlin/com/bugsnag/android/performance/internal/processing/SpanTaskWorker.kt

```diff
@@ -0,0 +1,111 @@
+package com.bugsnag.android.performance.internal.processing
+
+import android.os.SystemClock
+import com.bugsnag.android.performance.Logger
+
+import java.util.concurrent.DelayQueue
+import java.util.concurrent.Delayed
+import java.util.concurrent.TimeUnit
+
+internal interface ScheduledAction : Delayed, Runnable {
+    override fun compareTo(other: Delayed): Int {
+        val delay = getDelay(TimeUnit.NANOSECONDS)
+        val otherDelay = other.getDelay(TimeUnit.NANOSECONDS)
+        return when {
+            delay < otherDelay -> -1
+            delay > otherDelay -> 1
+            else -> 0
+        }
+    }
+}
+
+internal class SpanTaskWorker : Runnable, TimeoutExecutor {
+    /*
+     * The SpanTaskWorker is a Timer/ScheduledExecutor/HandlerThread style class, but allows tasks
+     * to be added and removed before the backing thread is actually started. Once it has been
+     * started any past tasks are executed immediately. This suits us nicely since our configuration
+     * only typically settles some time after the first actions have already been taken.
+     */
+
+    @Volatile
+    private var running = false
+
+    private var thread: Thread? = null
+
+    private val actions = DelayQueue<ScheduledAction>()
+
+    @Synchronized
+    fun start() {
+        if (running) {
+            return
+        }
+
+        running = true
+        thread = Thread(this, "Bugsnag Span Worker")
```

Comment:
Is it more descriptive to say Span Task Worker?
```suggestion
        thread = Thread(this, "Bugsnag Span Task Worker")
```